### PR TITLE
Pens for heads

### PR DESCRIPTION
**Pull request by justdie12**
logo-ru

## About the PR
Make some funny stuff for heads. I will make more, for clown or mime, but now i PR'd this.

## Why / Balance
In emergency everybody needs a pen! (QM is a businessman, he urgently needs money)

## Media
https://drive.google.com/file/d/1iEHP8b9V5CrUGO42u-YBavyTxxRc1ndM/view

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

🆑
- add: Command members now all have unique pens, each with their own unique features

### CODE REVIEW

**Comment by justdie12** [comment]
https://drive.google.com/file/d/1HRsDyp97hmL2DK1-Wf5W7R5CRk4OWekn/view?usp=sharing

**Comment by deltanedas** [comment]
clown gets luxury crayon instead of pen

**Comment by beck-thompson** [comment]
It would be nice if you wrote out the change along with the video!

**Comment by justdie12** [comment]
> It would be nice if you wrote out the change along with the video!

i do, but it's so big... 
https://drive.google.com/file/d/1HRsDyp97hmL2DK1-Wf5W7R5CRk4OWekn/view?usp=sharing
there

**Comment by ps3moira** [comment]
what would be the point of unique pen's for every single head 🤔

**Comment by KaiserMaus** [comment]
New Thief sleal objective?

**Comment by ArtisticRoomba** [comment]
Pen thief objective would be a good excuse to have cool custom pens for each head.

I like it, but my only gripe is that the art style of the pens clashes with the current pens (HoP and Cap) in the game. They seem too thick.

**Comment by UbaserB** [comment]
this is kind of weird

**Comment by justdie12** [comment]
> Pen thief objective would be a good excuse to have cool custom pens for each head.
> 
> I like it, but my only gripe is that the art style of the pens clashes with the current pens (HoP and Cap) in the game. They seem too thick.

ooookay, i go to re-sprite

**Comment by pheenty** [comment]
I'd remove self-recharging from everyone but CE

**Comment by justdie12** [comment]
> I'd remove self-recharging from everyone but CE

they very slow self-recharges itself in cause of emergency. 10 minutes for self-recharging, kinda slow?

**Comment by justdie12** [comment]
Resprited and reviewed. Added some cool stuff like animation

**Comment by Plykiya** [comment]
> > It would be nice if you wrote out the change along with the video!
> 
> i do, but it's so big... https://drive.google.com/file/d/1HRsDyp97hmL2DK1-Wf5W7R5CRk4OWekn/view?usp=sharing there

a changelog for this kind of relatively large content PR is going to be necessary, especially when it affects game balance

**Comment by justdie12** [comment]
> balance feedback

image
image

**Comment by LankLTE** [comment]
I don’t personally think each head needs their own unique pen, but especially not ones that have side-effects that are this strong. From the looks of it the RD’s just completely invalidates the CHIMP research and the HoS’s is just a stronger baton. These are pens, they should be for writing.

**Comment by ps3moira** [comment]
I agree it's pretty unnecessary for individual pens for each and comes off bloaty.

**Comment by ArtisticRoomba** [comment]
Yeah after reading the diffs, it would be way better for these pens to not have abilities at all. I'm not going to reiterate what others have said, but it's just too much. 

I'd prefer it if they simply got a better looking pen with no abilities.

**Comment by UBlueberry** [comment]
this PR is slightly misleading, at face value it just seems like pen reskins but - surprise! - please find stun baton attached.

**Comment by LittleNorthStar** [comment]
If they're going to get unique abilities they need to be minor things which are already accomplished by other stuff
what I would do for example
CMO pen can be used for cauterizing injuries
RD pen can scan anomalies (make it take longer than a normal scanner if possible)
HOS pen can hold 5 units of a liquid (for the enterprising alcoholic or combat medicine)
CE pen can pulse wires like a multi tool
QM pen has integrated Accepted / Denied stamps

and since it would be only fair
Hop pen can check the access on a card (no editing)
captain pen can have a built in lighter (for burning all the documents before the IRS get in)

**Comment by justdie12** [comment]
> RD может сканировать аномалии (есл



> If they're going to get unique abilities they need to be minor things which are already accomplished by other stuff what I would do for example CMO pen can be used for cauterizing injuries RD pen can scan anomalies (make it take longer than a normal scanner if possible) HOS pen can hold 5 units of a liquid (for the enterprising alcoholic or combat medicine) CE pen can pulse wires like a multi tool QM pen has integrated Accepted / Denied stamps
> 
> and since it would be only fair Hop pen can check the access on a card (no editing) captain pen can have a built in lighter (for burning all the documents before the IRS get in)

i do some of this functions.

**Comment by beck-thompson** [comment]
I actually think the QMs ability is fine, it should just be a lower cap! I was thinking HOSes should be like low power flashlight or something.

**Comment by UBlueberry** [comment]
> If they're going to get unique abilities they need to be minor things which are already accomplished by other stuff what I would do for example CMO pen can be used for cauterizing injuries RD pen can scan anomalies (make it take longer than a normal scanner if possible) HOS pen can hold 5 units of a liquid (for the enterprising alcoholic or combat medicine) CE pen can pulse wires like a multi tool QM pen has integrated Accepted / Denied stamps
> 
> and since it would be only fair Hop pen can check the access on a card (no editing) captain pen can have a built in lighter (for burning all the documents before the IRS get in)

even this is too much IMO. my issue that is that the pens are too overpowered, it's that they shouldn't have power in the first place. they're _pens_. 
what if every head's flash got a "special feature"? would it make sense if the RD pulls out a flash to scan an anomaly instead of using an anomaly scanner?
if you think that's a bad idea, then why should a pen do the same? 
flashes are for self-defense. pens are for writing. let them stay that way, please

File: Resources/Textures/Objects/Misc/pens.rsi/meta.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/e1142f20f5e4661cb6845cfcf2dd69f864d67432. pen_centcom is a resprited version of pen_cap by PuroSlavKing (Github). Luxury pen is drawn by Ubaser.",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/e1142f20f5e4661cb6845cfcf2dd69f864d67432. pen_centcom is a resprited version of pen_cap by PuroSlavKing (Github). Luxury pen is drawn by Ubaser. Pens CE, RD, HOS, CMO, QM maded by justdie4908(Discord) and ghostovish7728(Discord)",
```
Comment (UBlueberry) [inline Resources/Textures/Objects/Misc/pens.rsi/meta.json:4]:
```suggestion
  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/e1142f20f5e4661cb6845cfcf2dd69f864d67432. pen_centcom is a resprited version of pen_cap by PuroSlavKing (Github). Luxury pen is drawn by Ubaser. CE, RD, HOS, CMO, and QM pens made by justdie4908(Discord) and ghostovish7728(Discord)",
```
grammar

**Comment by justdie12** [comment]
https://drive.google.com/file/d/1iEHP8b9V5CrUGO42u-YBavyTxxRc1ndM/view?usp=sharing
@Plykiya @UBlueberry what do you think about it now?

**Comment by LeIronGoat** [comment]
> even this is too much IMO. my issue that is that the pens are too overpowered, it's that they shouldn't have power in the first place. they're _pens_. what if every head's flash got a "special feature"? would it make sense if the RD pulls out a flash to scan an anomaly instead of using an anomaly scanner? if you think that's a bad idea, then why should a pen do the same? flashes are for self-defense. pens are for writing. let them stay that way, please

I don't think comparing a pen to a flash is equivalent. Also, spy-pens and similar have been a thing in media/life for absolute ages. (Pens with microphones, cameras, flashes, poison, _mind-wiping functionality_). Its not unreasonable to believe that a department head was given some sort of slightly modified pen as a cool gift (if it functionally replaces an existing round-start item the head would have anyway, it could be a cool loadout reward. CentCom's award for five years of service, a pen, hooray). _However_ I do believe that is a key point, the functionality for each should be extremely basic (and restricted to something the Head would otherwise have round-start) _**if any functionality is given at all.**_

To me, having it as a loadout reward would kind of fix that "why did they create pens with special functionality". Because its a "work reward". 

I'm not here to push for this change, I'm pretty indifferent. But I do think its perfectly fine to add small things like this. It adds a lot of charm _if done appropriately._ But notice I said small, because I do very much agree with these needing to be incredibly simple functions if they're to have functions at all.

Anyway, some quick thoughts if the functionality aspect is desired.

I like the idea of a multi-tool for CE. Let the pen short wires/contacts. Very available to engineering round start.
RD's could be a remote trigger. Very available to science round-start.
CMO's could be a labeler, vial-size chemical storage, or one-time emergency epi-pen.
HoS' could be a flash, either with 2/3 charges before permanently depleted, or refreshed with another flash. (Honestly, it'd be more annoying in my opinion to use it over just a normal flash anyway)
QM - Maybe if not the stamp idea, prince checker functionality?

Just my thoughts. I don't necessarily think they should have functions, but I do think there could be an appropriate level if it is desired.

**Comment by justdie12** [comment]
> I like the idea of a multi-tool for CE. Let the pen short wires/contacts. Very available to engineering round start. RD's could be a remote trigger. Very available to science round-start. CMO's could be a labeler, vial-size chemical storage, or one-time emergency epi-pen. HoS' could be a flash, either with 2/3 charges before permanently depleted, or refreshed with another flash. (Honestly, it'd be more annoying in my opinion to use it over just a normal flash anyway) QM - Maybe if not the stamp idea, price checker functionality?
> 
> Just my thoughts. I don't necessarily think they should have functions, but I do think there could be an appropriate level if it is desired.

So far, no moderator has requested changes or pen balances, I'll just wait. But if they do, I'll try to do everything you listed, except for the QM pen. By the way, the idea of a pen for HOS is still open, because none of our meager team has come up with decent functionality for it. Well, my idea was a 3-charge flash, and I think that's the best thing to give HOS

**Comment by Doc-Michael** [comment]
> > I like the idea of a multi-tool for CE. Let the pen short wires/contacts. Very available to engineering round start. RD's could be a remote trigger. Very available to science round-start. CMO's could be a labeler, vial-size chemical storage, or one-time emergency epi-pen. HoS' could be a flash, either with 2/3 charges before permanently depleted, or refreshed with another flash. (Honestly, it'd be more annoying in my opinion to use it over just a normal flash anyway) QM - Maybe if not the stamp idea, price checker functionality?
> > Just my thoughts. I don't necessarily think they should have functions, but I do think there could be an appropriate level if it is desired.
> 
> So far, no moderator has requested changes or pen balances, I'll just wait. But if they do, I'll try to do everything you listed, except for the QM pen(because it was my best idea for pen). By the way, the idea of a pen for HOS is still open, because none of our meager team has come up with decent functionality for it. Well, my idea was a 3-charge flash, and I think that's the best thing to give HOS

I'll offer an idea on what the HoS's pen could be.
when activated it turns into a small pen knife, dealing 7 slash, can be thrown for 10 slash,  simple idea.

**Comment by pheenty** [comment]
> > > I like the idea of a multi-tool for CE. Let the pen short wires/contacts. Very available to engineering round start. RD's could be a remote trigger. Very available to science round-start. CMO's could be a labeler, vial-size chemical storage, or one-time emergency epi-pen. HoS' could be a flash, either with 2/3 charges before permanently depleted, or refreshed with another flash. (Honestly, it'd be more annoying in my opinion to use it over just a normal flash anyway) QM - Maybe if not the stamp idea, price checker functionality?
> > > Just my thoughts. I don't necessarily think they should have functions, but I do think there could be an appropriate level if it is desired.
> > 
> > 
> > So far, no moderator has requested changes or pen balances, I'll just wait. But if they do, I'll try to do everything you listed, except for the QM pen(because it was my best idea for pen). By the way, the idea of a pen for HOS is still open, because none of our meager team has come up with decent functionality for it. Well, my idea was a 3-charge flash, and I think that's the best thing to give HOS
> 
> I'll offer an idea on what the HoS's pen could be. when activated it turns into a small pen knife, dealing 7 slash, can be thrown for 10 slash, simple idea.

Centcomm pen already does that

**Comment by UBlueberry** [comment]
> So far, no moderator has requested changes or pen balances, I'll just wait. 

I should make it clear that I'm not a moderator, just a guy with opinions.

> Also, spy-pens and similar have been a thing in media/life for absolute ages. Pens with microphones, cameras, flashes, poison, mind-wiping functionality... not to mention **uplinks in SS13**. Its not unreasonable to believe that a department head was given some sort of slightly modified pen as a cool gift (if it functionally replaces an existing round-start item the head would have anyway, it could be a cool loadout reward. CentCom's award for five years of service, a pen, hooray). 

I'm still of the belief that these pens should just be pens, just as the luxury pen is... a pen.
But I think this could make a good compromise.
Here's an idea: penlights? Small, roundstart, saves me the trouble of heading into maints, and doesn't really steal the spotlight from anything.
I also think it's funny. "Thanks for your commitment to Nanotrasen, here's a shiny in your pen."

**Comment by LeIronGoat** [comment]
> So far, no moderator has requested changes or pen balances, I'll just wait. But if they do, I'll try to do everything you listed, except for the QM pen(because it was my best idea for pen). By the way, the idea of a pen for HOS is still open, because none of our meager team has come up with decent functionality for it. Well, my idea was a 3-charge flash, and I think that's the best thing to give HOS

Oh, I definitely liked your QM idea, I thought it was funny. Just wanted to provide some some more "basic" functional ideas if people didn't like what was proposed.

**Comment by Plykiya** [comment]
> https://drive.google.com/file/d/1iEHP8b9V5CrUGO42u-YBavyTxxRc1ndM/view?usp=sharing @Plykiya @UBlueberry what do you think about it now?

Please write out a description of what the PR actually contains in the description rather than using a video, you can use bulletpoints to separate the content. Also update the changelog section to indicate what's been added, even if it is as simple as "add: Command members now all have unique pens, each with their own unique features."

File: Resources/Prototypes/Entities/Objects/Misc/pen.yml [hunk 2]
```diff
@@ -119,3 +119,172 @@
...
+  - type: MeleeWeapon
+    wideAnimationRotation: -135
+    damage:
+      types:
+        Piercing: 5
+    bluntStaminaDamageFactor: 2.0
+    angle: 60
+    animation: WeaponArcSlash
+  - type: StaminaDamageOnHit
+    damage: 100
```
[the remark this answers is not in the document]
Comment (Plykiya) [reply Resources/Prototypes/Entities/Objects/Misc/pen.yml]: Alright it's not a stunbaton anymore, it's a flash, so this comment isn't relevant anymore.

**Comment by justdie12** [comment]
> > https://drive.google.com/file/d/1iEHP8b9V5CrUGO42u-YBavyTxxRc1ndM/view?usp=sharing @Plykiya @UBlueberry what do you think about it now?
> 
> Please write out a description of what the PR actually contains in the description rather than using a video, you can use bulletpoints to separate the content. Also update the changelog section to indicate what's been added, even if it is as simple as "add: Command members now all have unique pens, each with their own unique features."

okay, but what about balance?

**Comment by Plykiya** [comment]
> okay, but what about balance?

nothing particularly I can say, you just need to wait for maintainers to officially review it at this point

**Comment by UBlueberry** [comment]
Gonna have to say the same here

**Comment by ps3moira** [comment]
These aren't normal pens these all do stuff that pre-existing items do but can be used to write. Command already has items that do certain actions these pens already do and some have to work for it i.e. researching tech to unlock the tech instead of being given it round start. Stylized pens for each head is one thing, giving them round-start devices that already exist/need to be earned guised as regular pens is another.

**Comment by slarticodefast** [comment]
Hey, I brought this up for maintainer discussion and we come to the conclusion not to merge this for the following reasons:
- Many of these abilities are already available as a normal item, there is no need or reason for them to be part of a pen as well.
- Some of these are a little unbalanced, for example the QM's pen is simply free cash.
- Even if we remove the abilities, adding them would fall more into the item bloat category.
- The special pens would make impersonating a department head harder unless we would also add a chameleon pen.
Thank you for your work.

**Pull request closed without merging**